lib/full-site-editing/template-loader.php

```diff
@@ -12,8 +12,8 @@ function gutenberg_add_template_loader_filters() {
 	if ( ! gutenberg_supports_block_templates() ) {
 		return;
 	}
-
-	foreach ( gutenberg_get_template_type_slugs() as $template_type ) {
+	$template_type_slugs = array_keys( get_default_block_template_types() );
+	foreach ( $template_type_slugs as $template_type ) {
 		if ( 'embed' === $template_type ) { // Skip 'embed' for now because it is not a regular template type.
 			continue;
 		}
```

Comment:
The template loader is something that landed on 5.8 I believe but was tweaked on 5.9, so ideally the code here should be moved somehow to `lib/compat/5.9` folder maybe. cc @ockham

Reply:
Yep! I see you're addressing that in #37149 👍